sys/net/nfc/ndef/ndef_message.c

```diff
@@ -0,0 +1,270 @@
+
+    /* chunking is not supported */
+    cf = false;
+
+    /* if the payload is short, set sr to true */
+    if (payload_length <= SHORT_RECORD_PAYLOAD_LENGTH) {
+        sr = true;
+    }
+    else {
+        sr = false;
+    }
+
+    /* if an ID is present, set il to true */
+    if (id) {
+        il = true;
+    }
+    else {
+        il = false;
+    }
+
+    uint8_t header_byte = 0;
+    header_byte |= mb ? RECORD_MB_MASK : 0;
+    header_byte |= me ? RECORD_ME_MASK : 0;
+    header_byte |= cf ? RECORD_CF_MASK : 0;
+    header_byte |= sr ? RECORD_SR_MASK : 0;
+    header_byte |= il ? RECORD_IL_MASK : 0;
+    header_byte |= tnf;
+
+    if (sr) {
+        record.record_type = NDEF_SHORT_RECORD;
+    }
+    else {
+        record.record_type = NDEF_LONG_RECORD;
+    }
+
+    record.header = ndef_write_to_buffer(message, &header_byte, 1);
+    record.start = record.header;
+    record.type_length = ndef_write_to_buffer(message, &type_length, 1);
+
+    if (record.record_type == NDEF_SHORT_RECORD) {
+        uint8_t payload_length_single_byte = (uint8_t)payload_length & 0xFF;
+        record.payload_length = ndef_write_to_buffer(message, (uint8_t *)&payload_length_single_byte, SHORT_RECORD_PAYLOAD_LENGTH_SIZE);
+    }
+    else {
+        #if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+            // payload length shall be MSB first or Big Endian - need to reverse if LE
+            uint32_t payload_length_multi_byte = htonl(payload_length);
+            record.payload_length = ndef_write_to_buffer(message, (uint8_t*) &payload_length_multi_byte, LONG_RECORD_PAYLOAD_LENGTH_SIZE);
+        #else        
+            record.payload_length = ndef_write_to_buffer(message, (uint8_t *)&payload_length, LONG_RECORD_PAYLOAD_LENGTH_SIZE);
+        #endif
+    }
+
+    if (id) {
+        record.id_length = ndef_write_to_buffer(message, &id_length, 1);
+    }
+
+    record.type = ndef_write_to_buffer(message, (uint8_t *)type, type_length);
+
+    if (id) {
+        record.id = ndef_write_to_buffer(message, id, id_length);
+    }
+
+    if (payload != NULL) {
+        record.payload = ndef_write_to_buffer(message, payload, payload_length);
+    }
+    else {
+        /** 
+         *  this needs to be done so the payload points to the correct position 
+         *  for further writes
+         */
+        record.payload = &message->buffer.memory[message->buffer.cursor];
+
+        /* the payload has to be written by the calling function */
+        (void)payload;
+    }
+
+    message->records[message->record_count] = record;
+    message->record_count += 1;
+
+    return 0;
+}
+
+int ndef_remove_record(ndef_t *message)
+{
+    assert(message != NULL);
+
+    if (message->record_count == 0) {
+        LOG_ERROR("No records to remove");
+        return -1;
+    }
+
+    /* for only one record we simply set the count and cursor to 0 */
+    if (message->record_count == 1) {
+        message->record_count = 0;
+        message->buffer.cursor = 0;
```

Comment:
```suggestion
        ndef_clear(message);
```

as it does the same, right?